app_legacy/Helpers/database/player-game.php

```diff
@@ -412,38 +412,65 @@ function getUsersCompletedGamesAndMax(string $user): array
     return legacyDbFetchAll($query)->toArray();
 }
 
-function getTotalUniquePlayers(int $gameID, ?string $requestedBy = null, bool $hardcoreOnly = false, ?int $achievementType = null): int
+function getTotalUniquePlayers(int $gameID, ?int $parentGameID = null, ?string $requestedBy = null, bool $hardcoreOnly = false, ?int $achievementType = null): int
 {
     $bindings = [
         'gameId' => $gameID,
     ];
 
-    $hardcoreStatement = '';
+    $baseUnlockModeStatement = '';
     if ($hardcoreOnly) {
-        $bindings['unlockMode'] = UnlockMode::Hardcore;
-        $hardcoreStatement = ' AND aw.HardcoreMode = :unlockMode';
+        $bindings['baseUnlockMode'] = UnlockMode::Hardcore;
+        $baseUnlockModeStatement = ' AND awh.HardcoreMode = :baseUnlockMode';
     }
 
-    $achievementStatement = '';
+    $baseAchievementFlagsStatement = '';
     if ($achievementType !== null) {
-        $bindings['achievementType'] = $achievementType;
-        $achievementStatement = 'AND ach.Flags = :achievementType';
+        $bindings['baseAchievementType'] = $achievementType;
+        $baseAchievementFlagsStatement = 'AND achh.Flags = :baseAchievementType';
     }
 
     $requestedByStatement = '';
     if ($requestedBy) {
         $bindings['requestedBy'] = $requestedBy;
-        $requestedByStatement = 'OR ua.User = :requestedBy';
+        $requestedByStatement = 'OR uah.User = :requestedBy';
+    }
+
+    $parentGameIdStatement = '';
+    $parentUnlockModeStatement = '';
+    $parentAchievementTypeStatement = '';
+    if ($parentGameID !== null) {
```

Comment:
I believe you can do the same kind of cleanup here not needing much of the extra parent variables or the union if you do `where gameid in (gameid, parentid)`

Reply:
Thanks to your suggestion I was able to trim this down significantly in https://github.com/RetroAchievements/RAWeb/pull/1530/commits/f367e723499103b84452ce04e5087506bb829f7c.

I am working through the achievement page issue now.